[MTM-64869] Change logs for Java SDK, bugfix for resolving tenant or … (#3805)
SoftwareAG/c8y-docs: content/change-logs/platform-services/cumulocity-2025-78-0-External-Jwt-token-load-correctly-tenant-and-username.md

---
date:
title: Improved Java SDK support for external IAM tokens without tenant or username claims
product_area: Application enablement & solutions
change_type:
  - value: change-2c7RdTdXo4
    label: Improvement
component:
  - value: QWPx3rFfn
    label: Java SDK
build_artifact:
  - value: tc-QHwMfWtBk7
    label: cumulocity
ticket: MTM-64869
version: 2025.78.0
---

> Resolved an issue in the Java SDK where authentication using [external IAM JWT tokens](/authentication/sso/#configuring-access-tokens) failed to correctly identify the tenant and user when the token did not include a tenant ID or username in its claims.

Reviewer: ```suggestion
If using [external IAM JWT tokens](/authentication/sso/#configuring-access-tokens), the authentication failed to correctly identify the tenant and user when the token did not include a tenant ID or username in its claims. 
```

> With this fix, the Java SDK now properly retrieves the tenant ID and username from the {{< product-c8y-iot >}} credentials context, ensuring correct identification and smoother integration with external identity providers.

Reviewer 2: maybe `With this improvement`? Was it a bug or "not implemented" ?

Author: fix :)

Reviewer: How about "with this change" :-)?

Author: ok, updated

> With this fix, the SDK now properly retrieves the tenantId and username from the CumulocityCredentials context, ensuring correct identification and smoother integration with external identity providers.

Reviewer: ```suggestion
With this fix, the Java SDK now properly retrieves the tenant ID and username from the {{< product-c8y-iot >}} credentials context, ensuring correct identification and smoother integration with external identity providers.
```

version: 2025.78.0
---

> If using [external IAM JWT tokens](/authentication/sso/#configuring-access-tokens), the authentication failed to correctly identify the tenant and user when the token did not include a tenant ID or username in its claims. 

Reviewer 3: change If to When

Author: Done
